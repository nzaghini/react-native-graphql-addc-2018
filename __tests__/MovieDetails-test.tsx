import React from "react";
import renderer from "react-test-renderer";
import MovieDetails from "../src/screens/MovieDetail";
import { mockNavigation } from "./mocks/NavigationMock";

const mockListOfMovies = [ 
    { id: "1", title: "Interstellar", year: "2014", director: { firstName: "Christopher", lastName: "Nolan" } },
    { id: "2", title: "Mad Max: Fury Road", year: "2015", director: { firstName: "George", lastName: "Miller" } },
];

describe("<MovieDetails /> with Movie provided via navigation", () => {
                      
    it("Renders correctly", () => {
        jest.spyOn(mockNavigation, "getParam").mockImplementation(({ screen }) => mockListOfMovies[0]);

        const tree = renderer.create(<MovieDetails navigation={mockNavigation} />).toJSON();
        expect(JSON.stringify(tree)).toContain(mockListOfMovies[0].title);
        expect(tree).toMatchSnapshot();
    });

});

describe("<MovieDetails /> without Movie provided via props directly", () => {    
                      
    it("Something wrong here", () => {
        jest.spyOn(mockNavigation, "getParam").mockImplementation(({ screen }) => {});
        
        const tree = renderer.create(<MovieDetails navigation={mockNavigation} movie={mockListOfMovies[1]} />).toJSON();
        expect(JSON.stringify(tree)).toContain(mockListOfMovies[1].title);
        expect(tree).toMatchSnapshot();
    });

});

describe("<MovieDetails /> without Movie provided", () => {    
                      
    it("Something wrong here", () => {
        jest.spyOn(mockNavigation, "getParam").mockImplementation(({ screen }) => {});

        const tree = renderer.create(<MovieDetails navigation={mockNavigation} />).toJSON();
        expect(JSON.stringify(tree)).toContain("Something wrong here!");
        expect(tree).toMatchSnapshot();
    });

});
