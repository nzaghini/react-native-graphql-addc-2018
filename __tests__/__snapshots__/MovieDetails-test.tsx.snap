// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDetails /> with Movie provided via navigation Renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 60,
        "padding": 5,
        "textAlign": "center",
      }
    }
  >
    Interstellar
  </Text>
  <Text
    style={
      Object {
        "color": "#a9a9a9",
        "fontSize": 20,
        "padding": 5,
        "textAlign": "center",
      }
    }
  >
    Director: Christopher Nolan
  </Text>
  <Text
    style={
      Object {
        "color": "#a9a9a9",
        "fontSize": 10,
        "padding": 5,
        "textAlign": "center",
      }
    }
  >
    Year: 2014
  </Text>
</View>
`;

exports[`<MovieDetails /> without Movie provided Something wrong here 1`] = `
<Text>
  Something wrong here!
</Text>
`;

exports[`<MovieDetails /> without Movie provided via props directly Something wrong here 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 60,
        "padding": 5,
        "textAlign": "center",
      }
    }
  >
    Mad Max: Fury Road
  </Text>
  <Text
    style={
      Object {
        "color": "#a9a9a9",
        "fontSize": 20,
        "padding": 5,
        "textAlign": "center",
      }
    }
  >
    Director: George Miller
  </Text>
  <Text
    style={
      Object {
        "color": "#a9a9a9",
        "fontSize": 10,
        "padding": 5,
        "textAlign": "center",
      }
    }
  >
    Year: 2015
  </Text>
</View>
`;
